From 956e7ffe26bc0c1c5ec4596f2468b0637913e474 Mon Sep 17 00:00:00 2001
From: Jared Davenport <duron27@gmail.com>
Date: Thu, 4 Jul 2024 18:04:02 +0000
Subject: [PATCH] Update CMakeLists.txt

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7466ea7593..700e7020e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -478,6 +478,7 @@ find_package(Boost 1.6.2 REQUIRED COMPONENTS ${BOOST_COMPONENTS} OPTIONAL_COMPON
 if(OPENMW_USE_SYSTEM_MYGUI)
     find_package(MyGUI 3.4.3 REQUIRED)
 endif()
+set(SDL2_NO_SDL2_CMAKE ON)
 find_package(SDL2 2.0.9 REQUIRED)
 find_package(OpenAL REQUIRED)
 find_package(ZLIB REQUIRED)
-- 
GitLab

