From ce5341c2b2bd1c1574b1db78f1585d7b24409fcd Mon Sep 17 00:00:00 2001
From: Duron27 <duron27@gmail.com>
Date: Thu, 4 Jul 2024 00:08:54 -0400
Subject: [PATCH] Update CMakeLists.txt

---
 CMakeLists.txt | 64 +++++++++++++++++++++++++++++++++-----------------
 1 file changed, 42 insertions(+), 22 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33b2b616fbbcb..8e53bdc738cf2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -531,28 +531,48 @@ endif()
 
 # General source files
 file(GLOB SOURCE_FILES
-  ${SDL2_SOURCE_DIR}/src/*.c
-  ${SDL2_SOURCE_DIR}/src/atomic/*.c
-  ${SDL2_SOURCE_DIR}/src/audio/*.c
-  ${SDL2_SOURCE_DIR}/src/cpuinfo/*.c
-  ${SDL2_SOURCE_DIR}/src/dynapi/*.c
-  ${SDL2_SOURCE_DIR}/src/events/*.c
-  ${SDL2_SOURCE_DIR}/src/file/*.c
-  ${SDL2_SOURCE_DIR}/src/joystick/*.c
-  ${SDL2_SOURCE_DIR}/src/haptic/*.c
-  ${SDL2_SOURCE_DIR}/src/hidapi/*.c
-  ${SDL2_SOURCE_DIR}/src/libm/*.c
-  ${SDL2_SOURCE_DIR}/src/locale/*.c
-  ${SDL2_SOURCE_DIR}/src/misc/*.c
-  ${SDL2_SOURCE_DIR}/src/power/*.c
-  ${SDL2_SOURCE_DIR}/src/render/*.c
-  ${SDL2_SOURCE_DIR}/src/render/*/*.c
-  ${SDL2_SOURCE_DIR}/src/sensor/*.c
-  ${SDL2_SOURCE_DIR}/src/stdlib/*.c
-  ${SDL2_SOURCE_DIR}/src/thread/*.c
-  ${SDL2_SOURCE_DIR}/src/timer/*.c
-  ${SDL2_SOURCE_DIR}/src/video/*.c
-  ${SDL2_SOURCE_DIR}/src/video/yuv2rgb/*.c)
+  ${SDL2_SOURCE_DIR}/src/*.c
+  ${SDL2_SOURCE_DIR}/src/audio/*.c
+  ${SDL2_SOURCE_DIR}/src/audio/android/*.c
+  ${SDL2_SOURCE_DIR}/src/audio/dummy/*.c
+  ${SDL2_SOURCE_DIR}/src/audio/aaudio/*.c
+  ${SDL2_SOURCE_DIR}/src/audio/openslES/*.c
+  ${SDL2_SOURCE_DIR}/src/atomic/SDL_atomic.c.arm
+  ${SDL2_SOURCE_DIR}/src/atomic/SDL_spinlock.c.arm
+  ${SDL2_SOURCE_DIR}/src/core/android/*.c
+  ${SDL2_SOURCE_DIR}/src/cpuinfo/*.c
+  ${SDL2_SOURCE_DIR}/src/dynapi/*.c
+  ${SDL2_SOURCE_DIR}/src/events/*.c
+  ${SDL2_SOURCE_DIR}/src/file/*.c
+  ${SDL2_SOURCE_DIR}/src/haptic/*.c
+  ${SDL2_SOURCE_DIR}/src/haptic/android/*.c
+  ${SDL2_SOURCE_DIR}/src/hidapi/*.c
+  ${SDL2_SOURCE_DIR}/src/hidapi/android/*.cpp
+  ${SDL2_SOURCE_DIR}/src/joystick/*.c
+  ${SDL2_SOURCE_DIR}/src/joystick/android/*.c
+  ${SDL2_SOURCE_DIR}/src/joystick/hidapi/*.c
+  ${SDL2_SOURCE_DIR}/src/joystick/virtual/*.c
+  ${SDL2_SOURCE_DIR}/src/loadso/dlopen/*.c
+  ${SDL2_SOURCE_DIR}/src/locale/*.c
+  ${SDL2_SOURCE_DIR}/src/locale/android/*.c
+  ${SDL2_SOURCE_DIR}/src/misc/*.c
+  ${SDL2_SOURCE_DIR}/src/misc/android/*.c
+  ${SDL2_SOURCE_DIR}/src/power/*.c
+  ${SDL2_SOURCE_DIR}/src/power/android/*.c
+  ${SDL2_SOURCE_DIR}/src/filesystem/android/*.c
+  ${SDL2_SOURCE_DIR}/src/sensor/*.c
+  ${SDL2_SOURCE_DIR}/src/sensor/android/*.c
+  ${SDL2_SOURCE_DIR}/src/render/*.c
+  ${SDL2_SOURCE_DIR}/src/render/*/*.c
+  ${SDL2_SOURCE_DIR}/src/stdlib/*.c
+  ${SDL2_SOURCE_DIR}/src/thread/*.c
+  ${SDL2_SOURCE_DIR}/src/thread/pthread/*.c
+  ${SDL2_SOURCE_DIR}/src/timer/*.c
+  ${SDL2_SOURCE_DIR}/src/timer/unix/*.c
+  ${SDL2_SOURCE_DIR}/src/video/*.c
+  ${SDL2_SOURCE_DIR}/src/video/android/*.c
+  ${SDL2_SOURCE_DIR}/src/video/yuv2rgb/*.c
+  ${SDL2_SOURCE_DIR}/src/test/*.c)
 
 
 set(SDL_DEFAULT_ASSERT_LEVEL_CONFIGURED 1)
