From 814a91c32a1977681229eb99d3a61caf644d6fa7 Mon Sep 17 00:00:00 2001
From: Duron27 <duron27@gmail.com>
Date: Fri, 19 Jul 2024 19:21:33 -0400
Subject: [PATCH] Update Swapchain.cpp

---
 src/vsg/vk/Swapchain.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/vsg/vk/Swapchain.cpp b/src/vsg/vk/Swapchain.cpp
index e38fa7888..bcd7470e6 100644
--- a/src/vsg/vk/Swapchain.cpp
+++ b/src/vsg/vk/Swapchain.cpp
@@ -207,7 +207,8 @@ Swapchain::Swapchain(PhysicalDevice* physicalDevice, Device* device, Surface* su
         createInfo.imageSharingMode = VK_SHARING_MODE_EXCLUSIVE;
     }
 
-    createInfo.preTransform = details.capabilities.currentTransform; 
+    createInfo.preTransform = VK_SURFACE_TRANSFORM_IDENTITY_BIT_KHR;
+    createInfo.preTransform = details.capabilities.currentTransform;
     createInfo.compositeAlpha = VK_COMPOSITE_ALPHA_OPAQUE_BIT_KHR;
     createInfo.presentMode = presentMode;
     createInfo.clipped = VK_TRUE;
