From c5fdad13e98ea0b146d5dfb1f5ed22c79c0f5379 Mon Sep 17 00:00:00 2001
From: Duron27 <duron27@gmail.com>
Date: Sun, 21 Jul 2024 14:36:53 -0400
Subject: [PATCH] supress warnings

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd88eaf0aeb..6fcffba2abd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -552,7 +552,7 @@ if(OPENMW_LTO_BUILD)
 endif()
 
 if (CMAKE_CXX_COMPILER_ID STREQUAL GNU OR CMAKE_CXX_COMPILER_ID STREQUAL Clang)
-    set(OPENMW_CXX_FLAGS "-Wall -Wextra -Wundef -Wextra-semi -Wno-unused-parameter -pedantic -Wno-long-long -Wnon-virtual-dtor -Wunused -Wno-mismatched-tags ${OPENMW_CXX_FLAGS}")
+    set(OPENMW_CXX_FLAGS "-Wall -Wextra -Wno-undef -Wno-unused-parameter -Wno-c++98-compat-extra-semi -Wno-long-long -Wnon-virtual-dtor -Wunused -Wno-overloaded-virtual -Wno-inconsistent-missing-override -Wno-deprecated-copy-with-user-provided-copy ${OPENMW_CXX_FLAGS}")
     add_definitions( -DBOOST_NO_CXX11_SCOPED_ENUMS=ON )
 
     if (APPLE)
